int32 index
bool ready